subsys/bluetooth/host/hci_core.c

```diff
@@ -3387,6 +3387,17 @@ void hci_event_prio(struct net_buf *buf)
 	}
 }
 
+#if !defined(CONFIG_BT_RECV_IS_RX_THREAD)
+static void rx_queue_put(struct net_buf *buf)
+{
+	net_buf_put(&bt_dev.rx_queue, buf);
+
+	if (!k_work_is_pending(&rx_work)) {
+		k_work_submit(&rx_work);
+	}
```

Comment:
We should check the return code of k_work_submit and log errors. 

The check for `k_work_is_pending` is not needed, as `k_work_submit` should handle that itself:

```c
 * @retval 0 if work was already submitted to a queue
 * @retval 1 if work was not submitted and has been queued to @p queue
 * @retval 2 if work was running and has been queued to the queue that was
 * running it
```

```suggestion
	int err;
	
	net_buf_put(&bt_dev.rx_queue, buf);

	err = k_work_submit(&rx_work); 
	if (err < 0) {
		BT_ERR("Could not submit rx_work: %d", err);
	}
```